test/pkg/firewall/firewall.go

```diff
@@ -0,0 +1,152 @@
+package firewall
+
+import (
+	"fmt"
+	"log"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+)
+
+// Apply the firewall rules on the node.
+func ApplyRulesToNode(NFTtable []byte, nodeName, artifactsDir string, utilsHelpers utils.UtilsInterface) error {
+	debugPod, err := utilsHelpers.CreatePodOnNode(nodeName, consts.DefaultDebugNamespace, consts.DefaultDebugPodImage)
+	if err != nil {
+		return fmt.Errorf("failed to create debug pod on node %s: %w", nodeName, err)
+	}
+
+	defer func() {
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			log.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	// save the nftables on  /host/etc/nftables/firewall.nft
+	_, err = RunCommandInPod(debugPod, fmt.Sprintf("echo '%s' > /host/etc/nftables/firewall.nft", string(NFTtable)), false, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to save rule set to file on node %s: %w", nodeName, err)
+	}
+
+	// run apply nft command for the rules
+	_, err = RunCommandInPod(debugPod, "nft -f /etc/nftables/firewall.nft", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to apply rule set on node %s: %w", nodeName, err)
+	}
+
+	output, err := nftList(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	// edit the sys conf to make sure it will be on the list after reboot
+	err = editNftablesConf(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	_, err = RunCommandInPod(debugPod, "/usr/sbin/nft list ruleset > /etc/nftables.conf", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", nodeName, err)
+	}
+
+	_, err = RunCommandInPod(debugPod, "systemctl enable nftables", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to enable nftables on node %s: %w", nodeName, err)
+	}
+
+	err = utilsHelpers.WriteFile(filepath.Join(artifactsDir, "nftables-"+nodeName), output)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func RulesList(nodeName, artifactsDir string, utilsHelpers utils.UtilsInterface) ([]byte, error) {
```

Comment:
`RulesList` is a general name to a function that can only be called after reboot since you set the that in the file name, please consider using making this func and `nftList` a single func that can be reused